### PR TITLE
fix: open specific nav group on docs page on first page load

### LINKED ISSUE #3543: [🕹️] Direct Link in Docs don't open Navigation Submenu
### Issue Summary

When opening a direct link to the docs, e.g. https://formbricks.com/docs/app-surveys/user-identification
the sub navigation menu on the left is closed
Screenshot 2024-10-12 at 12 12 41

even though the `User Idenification` page is part of the `Features` of Website & App surveys.

### Steps to Reproduce

1. Open a direct link for a subpage in the docs (e.g. https://formbricks.com/docs/app-surveys/user-identification )
2. Note that the submenu in the navigation that contains this page does not open automatically.

### Expected behavior

Make sure that when following a direct link opening the docs the right submenu in the navigation is opened automatically.
In case of the user identification page, the sub navigation should look like this:

Screenshot 2024-10-12 at 12 10 11

### Other information

_No response_

### Screenshots

_No response_

### Environment

- [ ] Formbricks Cloud (app.formbricks.com)
- [ ] Self-hosted Formbricks

### Desktop (please complete the following information)

```markdown
- OS:
- Node:
- npm:
```

### PR DESCRIPTION
**Pull request by DivyanshuLohani**
## What does this PR do?

This PR fixes the issue where the on direct navigation links do docs pages does not open the submenus where the section is present
Fixes #3543 

https://github.com/user-attachments/assets/bb538355-791b-4516-bf9c-b0cb9f63d4c5

## How should this be tested?

- Go on to the any docs page listed under a group
- Copy the url 
- Open it in another window

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [x] First PR at Formbricks? [Please sign the CLA!](https://cla-assistant.io/formbricks/formbricks) Without it we wont be able to merge it 🙏

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [x] Updated the Formbricks Docs if changes were necessary

## Summary by CodeRabbit

- **New Features**
	- Enhanced navigation experience by automatically expanding relevant sections based on the current page.
- **Bug Fixes**
	- Improved initial page load behavior for the navigation component.

### CODE REVIEW

File: apps/docs/components/Navigation.tsx [hunk 1]
```diff
@@ -160,6 +160,26 @@ const NavigationGroup = ({
...
+              setActiveGroup(group);
+              return true;
+            }
+          });
+        }
+      });
+    });
+    // We do not need any dependencies this should run on page load only
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```
[the remark this answers is not in the document]
Comment (jobenjada) [reply apps/docs/components/Navigation.tsx:181]: @DivyanshuLohani wanna implement that feedback? Makes sense to me!

**Review by jobenjada — CHANGES_REQUESTED** [review]
pls implement code rabbits suggestions

you'll get 300: 150 for spotting and 150 for fixing it

**Comment by DivyanshuLohani** [comment]
Done

**Review by jobenjada — APPROVED** [review]
thanks :)

**Pull request merged**